featurebyte/tile/tile_cache.py

```diff
@@ -384,24 +383,18 @@ async def _register_working_table(
                         f"REQ.{quoted_identifier(serving_name).sql()} <=> {table_alias}.{quoted_identifier(key).sql()}"
                     )
                 )
-            join_conditions.append(
-                expressions.LTE(
-                    this=last_tile_start_date_expr,
-                    expression=expressions.Identifier(
-                        this=f"{table_alias}.{InternalName.TILE_LAST_START_DATE}"
-                    ),
-                )
-            )
+            # Note: join_conditions is empty list if there is no entity column. In this case, there
+            # is only one row in the tracking table and the join condition can be omitted.
             table_expr = table_expr.join(
                 tracker_table_name,
                 join_type="left",
                 join_alias=table_alias,
-                on=expressions.and_(*join_conditions),
+                on=expressions.and_(*join_conditions) if join_conditions else None,
```

Comment:
Curious whether we have existing test case cover the case where `join_conditions` is empty list?

Reply:
Yes, this test covers that scenario: https://github.com/featurebyte/featurebyte/blob/e9b628e40c511383e70dec55cd165341658d291d/tests/integration/api/test_change_view_operations.py#L69

The test was failing without the special handling here.